### PR TITLE
Close response bodies

### PR DESCRIPTION
**Pull request by gary-kim**
Signed-off-by: Gary Kim <gary@garykim.dev>

### CODE REVIEW

File: room/room.go [hunk 1]
```diff
@@ -130,6 +131,7 @@ func (t *TalkRoom) ReceiveMessages(ctx context.Context) (chan ocs.TalkRoomMessag
 			if res.StatusCode == 200 {
 				lastKnown = res.Header.Get("X-Chat-Last-Given")
 				data, err := ioutil.ReadAll(res.Body)
+				_ = res.Body.Close()
```
Comment (tilosp) [inline room/room.go:134]: shouldn't we also close it if `res.StatusCode != 200`?

**Review by tilosp — CHANGES_REQUESTED** [review]
I don't think that the `Close()` calls are necessary for the cases where `.Do()` is used since it already does so internally, see: https://github.com/gary-kim/go-nc-talk/blob/e618cba1d7f1509352d65d4cfc4d9e7883566857/vendor/github.com/monaco-io/request/request.go#L14-L30

So it should only be necessary in the one case that `.Resp()` is used.

**Comment by gary-kim** [comment]
Ah, okay, let me remove it from everything but the `Resp()`

File: room/room.go [hunk 2]
```diff
@@ -127,6 +127,7 @@ func (t *TalkRoom) ReceiveMessages(ctx context.Context) (chan ocs.TalkRoomMessag
 			if err != nil {
 				continue
 			}
+			_ = res.Body.Close()
 			if res.StatusCode == 200 {
 				lastKnown = res.Header.Get("X-Chat-Last-Given")
 				data, err := ioutil.ReadAll(res.Body)
```
Comment (tilosp) [inline room/room.go:133]: the body is now used after being closed

Comment (gary-kim) [reply room/room.go:133]: :facepalm: That was why I put in in there.

**Review by tilosp — APPROVED** [review]

**Pull request merged**